src/tests/ftest/util/file_utils.py

```diff
@@ -0,0 +1,251 @@
+                     raise_exception=True, sudo=False, owner=None):
+    """Copy the source to the destination on each of the specified hosts.
+
+    Optionally (by default) ensure the destination directory exists on each of
+    the specified hosts prior to copying the source.
+
+    Args:
+        hosts (NodeSet): hosts on which to copy the source
+        source (str): the file to copy to the hosts
+        destination (str): the host location in which to copy the source
+        mkdir (bool, optional): whether or not to ensure the destination
+            directory exists on hosts prior to copying the source. Defaults to
+            True.
+        timeout (int, optional): command timeout. Defaults to 60 seconds.
+        verbose (bool, optional): whether to log the command run and
+            stdout/stderr. Defaults to True.
+        raise_exception (bool, optional): whether to raise an exception if the
+            command returns a non-zero exit status. Defaults to True.
+        sudo (bool, optional): whether to run the command via sudo. Defaults to
+            False.
+        owner (str, optional): if specified the owner to assign as the owner of
+            the copied file. Defaults to None.
+
+    Raises:
+        CommandFailure: if there is an error running the command and an exception was requested
+
+    Returns:
+        RemoteCommandResult: a grouping of the command results from the same hosts with the same
+            return status
+    """
+    result = None
+    if mkdir:
+        result = create_directory(
+            hosts, os.path.dirname(destination), verbose=verbose, raise_exception=raise_exception)
+    if result.passed:
+        if sudo:
+            # In order to copy a protected file to a remote host in CI the source will first be
+            # copied as is to the remote host
+            localhost = gethostname().split(".")[0]
+            other_hosts = NodeSet.fromlist([host for host in hosts if host != localhost])
+            if other_hosts:
+                # Existing files with strict file permissions can cause the subsequent non-sudo
+                # copy to fail, so remove the file first
+                _get_result(other_hosts, f'rm -f {source}', timeout, verbose, True)
+                result = distribute_files(
+                    other_hosts, source, source, mkdir=True,
+                    timeout=timeout, verbose=verbose,
+                    raise_exception=raise_exception, sudo=False, owner=None)
+            if result.passed:
+                # Then a local sudo copy will be executed on the remote node to copy the source
+                # to the destination
+                log = getLogger()
+                command = get_clush_command(
+                    hosts, args="-S -v", command="cp {} {}".format(source, destination),
+                    command_sudo=True)
+                result = run_command(log, command, verbose, timeout)
```

Comment:
Why not run_remote?

Reply:
We need to sudo copy

Reply:
As we talked about, this maybe could just use run_remote, but this PR just moves the definition of this function so okay to keep as-is either way